### ISSUE TITLE
linter: No source value in Checkstyle report

### ISSUE THREAD

**Issue by TheGuit** [oxc-project/oxc#10204]
### What version of Oxlint are you using?

0.16.4

### What command did you run?

npx oxlint@latest --tsconfig='tsconfig.json' --format=checkstyle .

### What does your `.oxlintrc.json` config file look like?

```jsonc
{
    "extends": [
        "plugin:import/typescript",
        "plugin:@typescript-eslint/recommended",
        "plugin:unicorn/all"
    ]
}
```

### What happened?

Hi,

When using oxlint to generate a report file at Checkstyle XML format (which is the format with information that I need), source property is empty :

```xml
    <file name="apps/example/example.controller.spec.ts">
        <error line="12" column="9" severity="warning"
            message="Variable &apos;logger&apos; is assigned a value but never used. Unused variables should start with a &apos;_&apos;."
            source="" />
    </file>
```

I try to understand the rust code, and I understand that normally it should contain rule_id.
